### PR TITLE
add a method to extract directly the waveform from PSL

### PR DESCRIPTION
**Pull request by tdixon97**
- [x] Add `waveform_from_pulse_shape_library()` function and supporting `_get_psl_waveforms_impl()` to extract full waveforms from PSL
- [x] Make the `t0` field optional in `get_hpge_pulse_shape_library()`
- [x] Add phi-angle support to waveform extraction
- [x] Fix unit consistency when t0 is absent
- [x] Fix LGDO mutation (avoid setting attrs on disk objects)
- [x] Fix phi fallback when phi is not provided for phi-enabled library
- [x] Add docs for the new functionality
- [x] Add improved test assertions (value checks for empty array case)
- [x] Add additional tests for better coverage:
  - [x] Test mismatched input shapes raise ValueError
  - [x] Test mismatched phi shape raises ValueError
  - [x] Test zero energy depositions produce zero waveform
  - [x] Test single event with single hit
  - [x] Test out-of-bounds r/z values are clamped to grid boundary

### CODE REVIEW

File: docs/source/manual/psd.md [hunk 1]
```diff
@@ -52,3 +66,49 @@ Finally, _reboost_ contains methods to account for the delayed charge collection
...
+
+:::{note}
+
+Currently n+ effects can only be included with the fixed template option of {func}`reboost.hpge.psd.maximum_current`.
+
+:::
+
+## Waveform simulation
+
+The library of simulated pulse shapes can also be used to extract a simulated waveform per event (see {func}`reboost.hpge.psd.waveform_from_pulse_shape_library`). This is simply performed by summing the waveforms in the pulse shape library weighted by their
```
Comment (SalehGiovanna) [inline docs/source/manual/psd.md:78]: Worth mentioning that if steps in a MC event are closer than the PSL grid size, the resulting waveform will carry no spatial information (effectively point-like)?

File: docs/source/manual/psd.md [hunk 2]
```diff
@@ -52,3 +66,49 @@ Finally, _reboost_ contains methods to account for the delayed charge collection
...
+energy.
+
+:::{warning}
+
+The pulse shape library should consist of waveforms aligned by t0 for waveform simulation.
```
Comment (SalehGiovanna) [inline docs/source/manual/psd.md:83]: I think it might be misleading that the A/E functions require tAmax-aligned waveforms (so no operation needed, as it is the original alignment in the PSL) while here we want t0-aligned (so a shift is required).
Maybe we can integrate this shift in this function? So the original PSL can be passed?

File: src/reboost/hpge/psd.py [hunk 3]
```diff
@@ -910,3 +1020,84 @@ def maximum_current(
...
+    phi
+        Optional azimuthal angle in degrees for each step. If provided and the pulse shape library
+        contains phi-dependent templates, the appropriate template is selected based on 45-degree
+        repeating symmetry.
+
+    Returns
+    -------
+    waveforms : awkward.Array
+        Awkward array of waveforms with shape ``(n_events, n_time_samples)``.
+        Each waveform has the same sampling and units as specified by the
```
Comment (SalehGiovanna) [inline src/reboost/hpge/psd.py:1069]: Not really same units as the current waveforms in the library? After weighted sum of the PSL current pulses, there is now an extra energy (keV) factor, so the relative scale of the resulting waveforms is correct, because all the waveforms in the PSL were simulated at the same energy, but the absolute scale is not comparable with the PSL waveforms and with data, I think

File: docs/source/manual/psd.md [hunk 4]
```diff
@@ -52,3 +66,49 @@ Finally, _reboost_ contains methods to account for the delayed charge collection
...
+
+```python
+# create the library
+library = reboost.hpge.utils.get_hpge_pulse_shape_library(...)
+
+# read the remage file
+steps = lh5.read("stp", ...).view_as("ak", with_units=True)
+
+# compute r, here x0, y0, z0 are the detector origin
+r = np.sqrt((steps.xloc - x0) ** 2 + (steps.yloc - y0) ** 2)
```
Comment (SalehGiovanna) [inline docs/source/manual/psd.md:97]: Worth showing also explicit calculation of phi in the detector frame for the steps? This also needs to be done by hand to pass phi argument
